rust/xaynet-analytics/src/database/analytics_event/repo.rs

```diff
@@ -0,0 +1,61 @@
+use isar_core::error::IsarError;
+
+use crate::database::{
+    analytics_event::data_model::AnalyticsEvent,
+    common::{IsarAdapter, Repo},
+    instance::DbInstance,
+};
+
+pub struct AnalyticsEventRepo {
+    collection_name: &'static str,
+    db: &'static DbInstance,
+}
+
+impl AnalyticsEventRepo {
+    pub fn new(collection_name: &'static str, db: &'static DbInstance) -> Self {
+        Self {
+            collection_name,
+            db,
+        }
+    }
+}
+
+impl Repo<AnalyticsEvent> for AnalyticsEventRepo {
+    fn add(&self, event: AnalyticsEvent) -> Result<(), IsarError> {
+        let mut object_builder = match self.db.get_object_builder(self.collection_name) {
```

Comment:
Here we can do `self.db.get_object_builder(...)?`

Reply:
done